<workflow id="CREATE_CLIENT_BY_ID" name="Создание карточки клиента по идентификационным данным">
    <event name="onCandidates">
        <![CDATA[def depIn(dep):
    from colvir_cbs.services import bank
    return po['app']['depCode'] in bank.getDepartmentsCodes(dep)
]]>
    </event>
    <script id="f59a5c3b-e7f4-429e-afa0-765e85c37cd6" name="Поиск клиента">
        <![CDATA[
request = {
    'pinfl': pv.get('pinfl'),
    'inn': pv.get('inn'),
    'nibbd': pv.get('nibbd'),
}

try:
    client = execObjectMethod({
        'object': 'client', 'method': 'identifyClient',
        'params': request
    })
except Exception as e:
    raise UserException({
        'message': 'Ошибка идентификации клиента',
        'description': 'Запрос:\n%s' % json.dumps(request, indent=4, ensure_ascii=False)
    })

pv['identifiedClient'] = None
if client:
    pv['identifiedClient'] = client['code']
    lv['identifiedClientName'] = client['name']]]>
    </script>
    <split id="5534bb32-cc13-4bbb-b816-28cce1341da3" name="Клиент существует">
        <condition id="294d4b37-dc40-4c26-9a70-161022596213" name="Да" expression="pv[&apos;identifiedClient&apos;] is not None">
            <script id="653061b6-87ba-4e2f-a893-68757383c404" name="Журнализация">
                <journal time="leave">
                    <![CDATA[Найдена карточка клиента, {% if pv.clientType == "fiz" %} ПИНФЛ: {{pv.pinfl}}{% elif pv.clientType == "ie" %}ПИНФЛ: {{pv.pinfl}}, ИНН: {{pv.inn}}{% else %}ИНН: {{pv.inn}}{% endif %} , код {{pv.identifiedClient}}, {{lv.identifiedClientName}}]]>
                </journal>
            </script>
        </condition>
        <condition id="b67c31d9-85bd-4830-9059-6e90e9c227a7" name="Нет">
            <timer id="ab67e052-4e79-4d8b-b82e-f4f059b80b4f" name="Создание карточки клиента" description="Создание карточки клиента" duration="00:00:00" errorDuration="00:05:00" action="continue"/>
            <split id="7e6c6102-97a9-45ba-867d-c63b01526dfa" name="Вид клиента">
                <condition id="3d98f7b8-a1fc-47a6-b6b0-b02e0dab5e1d" name="Физическое лицо" expression="pv[&apos;clientType&apos;] == &apos;fiz&apos;">
                    <script id="23777b72-fb32-4ed8-ba81-cbbe614fc071" name="Создание карточки физического лица">
                        <![CDATA[pv['last_error'] = None
try:
    request = {
        'pinfl': pv.get('pinfl'),
        'series': pv.get('series'),
        'number': pv.get('number'),
        'birthday': pv.get('birthday'),
        'inn': pv.get('inn'),
        'nibbd': pv.get('nibbd'),
    }
    response = execObjectMethod({
        'object': 'client', 'method': 'createClientByIdFiz_2',
        'params': {
            'client': request
        }
    })
    lv['client'] = response
    pv['identifiedClient'] = lv['client']['code']
    
    if response['errCode'] != '0':
        pv['last_error'] = {
            'message': 'Карточка клиента не открыта',
            'description': 'Код ошибки: %s, %s' % (response['errCode'], response['errMsg'])
        }
    
    
    token.save_vars({
        'request': request,
        'response': response,
    })
except UserException as e:
    pv['last_error'] = e.getError()
#raise UserException('Создание карточки клиента в разработке')]]>
                    </script>
                    <split id="74367784-92a1-40a1-b859-de7755853e1e" name="Карточка создана">
                        <condition id="3a9fca85-ea22-48c8-908c-3c0df134153f" name="Успешно создана" expression="pv[&apos;last_error&apos;] is None">
                            <script id="eded4d09-1359-4312-a420-36a69b4afa41" name="Журнализация">
                                <journal time="leave">
                                    <![CDATA[Создана карточка клиента {{lv.client.code}} для ПИНФЛ {{pv.pinfl}}]]>
                                </journal>
                            </script>
                        </condition>
                        <condition id="3660bb33-f375-490e-961a-ef7d830bf346" name="Ошибка создания карточки" expression="pv[&apos;last_error&apos;] is not None">
                            <task id="0bc214de-cf07-401c-b3b7-9682d0283ed6" name="Ошибка создания карточки клиента" autoStart="true" description="Ошибка создания карточки клиента, ПИНФЛ: {{pv.pinfl}}">
                                <event name="onPrepare">
                                    <![CDATA[
lv['error'] = pv['last_error']

lv['actions'] = []


lv['actions'].append(
    {'name': 'Повторить', 'value': 'repeat'}
)

lv['actions'].append(
    {'name': 'Отменить обработку', 'value': 'cancel'}
)
]]>
                                </event>
                                <clientTask path="/easyflow/error-dialog" taskName="{{task.description}}">
                                    <p name="caption" expression="task.get(&apos;description&apos;)"/>
                                    <p name="error" expression="lv[&apos;error&apos;]"/>
                                    <p name="actionVar" expression="&apos;user_action&apos;"/>
                                    <p name="actions" expression="lv[&apos;actions&apos;]"/>
                                    <p name="objectKey" expression="pv.get(&apos;objectKey&apos;)"/>
                                </clientTask>
                            </task>
                            <split id="6e318ff0-ed30-40a7-8825-f42be8395730" name="Выбор пользователя">
                                <condition id="9f3e27f9-e884-45cc-ac11-bae4a9c0ab88" name="Повторить" expression="pv[&apos;user_action&apos;] == &apos;repeat&apos;">
                                    <jump id="6393646f-7c31-4315-bebb-303034644512" name="Переход к Создание карточки физического лица" target_id="23777b72-fb32-4ed8-ba81-cbbe614fc071"/>
                                </condition>
                                <condition id="83c5735f-4dea-4baf-bdfd-b0f9a2d1145e" name="Отменить" expression="pv[&apos;user_action&apos;] == &apos;cancel&apos;">
                                    <script id="9f64fbd8-3bf9-4f7d-babb-c1b8ff985e02" name="Обработка отмены">
                                        <journal time="leave">
                                            <![CDATA[Отмена создания карточки клиента из за ошибок]]>
                                        </journal>
                                        <![CDATA[pv['result'] = {
    'status': 'cancel'
}]]>
                                    </script>
                                    <terminate id="2e9acfbc-6968-437e-af2d-1ce81e1db492" name="Завершение"/>
                                </condition>
                            </split>
                        </condition>
                    </split>
                </condition>
                <condition id="0a0982b1-021b-426b-808a-9eba859b1234" name="Юридическое лицо/ИП" expression="pv[&apos;clientType&apos;] in (&apos;jur&apos;, &apos;ie&apos;)">
                    <script id="bb482ff4-1aa9-4cc9-b594-3945ecc1d986" name="Создание карточки юридического лица">
                        <![CDATA[pv['last_error'] = None
try:
    request = {
        'inn': pv.get('inn'),
        'pinfl': pv.get('pinfl'),
        'nibbd': pv.get('nibbd'),
        'clientType': pv.get('clientType'),
    }
    response = execObjectMethod({
        'object': 'client', 'method': 'createClientByIdJur',
        'params': {
            'client': request
        }
    })

    token.save_vars({
        'request': request,
        'response': response,
    })

    """
        0 - карточка открыта
        1 - карточка в заявке
        3 - карточка закрыта
    """

    lv['response'] = response
    pv['identifiedClient'] = response['code']
    
    if response['errCode'] != '0':
        pv['last_error'] = {
            'message': 'Карточка клиента не открыта',
            'description': 'Код ошибки: %s, %s' % (response['errCode'], response['errMsg'])
        }
        
    
except UserException as e:
    pv['last_error'] = e.getError()

#raise UserException('Создание карточки клиента в разработке')
]]>
                    </script>
                    <split id="f71ca9a7-405f-4d07-ab51-4006376297cb" name="Карточка создана">
                        <condition id="b0f9b820-4662-4b41-8e41-152abfab1db7" name="Карточка создана" expression="pv[&apos;last_error&apos;] is None">
                            <script id="528fd313-f6a2-46d2-806e-dfe96cf01f68" name="Журнализация">
                                <journal time="leave">
                                    <![CDATA[Создана карточка клиента {{lv.client.code}} для НИББД {{pv.nibbd}}]]>
                                </journal>
                            </script>
                        </condition>
                        <condition id="3fad3c7e-2279-4d44-8f09-0bad3e3667a8" name="Ошибка создания карточки" expression="pv[&apos;last_error&apos;] is not None">
                            <task id="8272240e-3900-436c-b92c-d7e99bcc7abc" name="Ошибка создания карточки клиента" autoStart="true" description="Ошибка создания карточки клиента, ИНН: {{pv.inn|default_if_none:&quot;&quot;}}, ПИНФЛ:  {{pv.pinfl|default_if_none:&quot;&quot;}}, НИББД: {{pv.nibbd|default_if_none:&quot;&quot;}}">
                                <event name="onPrepare">
                                    <![CDATA[
lv['error'] = pv['last_error']


lv['actions'] = []


lv['actions'].append(
    {'name': 'Повторить', 'value': 'repeat'}
)

lv['actions'].append(
    {'name': 'Отменить обработку', 'value': 'cancel'}
)
]]>
                                </event>
                                <clientTask path="/easyflow/error-dialog" taskName="{{task.description}}">
                                    <p name="caption" expression="task.get(&apos;description&apos;)"/>
                                    <p name="error" expression="lv[&apos;error&apos;]"/>
                                    <p name="actionVar" expression="&apos;user_action&apos;"/>
                                    <p name="actions" expression="lv[&apos;actions&apos;]"/>
                                    <p name="objectKey" expression="pv.get(&apos;objectKey&apos;)"/>
                                </clientTask>
                            </task>
                            <split id="5550e1d9-044f-4067-a5ac-8af8913d4a5d" name="Выбор пользователя">
                                <condition id="6d2991fb-0e04-4756-9511-7ad932e82a57" name="Повторить" expression="pv[&apos;user_action&apos;] == &apos;repeat&apos;">
                                    <jump id="2fb1a1ea-847f-4463-b36a-f92b8be92486" name="Переход к Создание карточки юридического лица" target_id="bb482ff4-1aa9-4cc9-b594-3945ecc1d986"/>
                                </condition>
                                <condition id="a5d57d27-5a7d-49ee-9470-458ab5c514d6" name="Отменить" expression="pv[&apos;user_action&apos;] == &apos;cancel&apos;">
                                    <script id="bc5a23a8-cb5b-4c10-ba44-cbaa4a39e676" name="Обработка отмены">
                                        <journal time="leave">
                                            <![CDATA[Отмена создания карточки клиента из за ошибок]]>
                                        </journal>
                                        <![CDATA[pv['result'] = {
    'status': 'cancel'
}]]>
                                    </script>
                                    <terminate id="52a36576-d560-4fd0-bf0e-bc734d18d5ac" name="Завершение"/>
                                </condition>
                            </split>
                        </condition>
                    </split>
                </condition>
            </split>
        </condition>
    </split>
    <script id="686a514c-787d-4704-83ce-50c53082b9f1" name="Установка результата: успешно">
        <![CDATA[pv['result'] = {
    'status': 'success'
}]]>
    </script>
</workflow>